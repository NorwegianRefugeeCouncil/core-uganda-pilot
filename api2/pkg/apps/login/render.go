package login

import "net/http"

func (s *Server) Render(w http.ResponseWriter, req *http.Request, templateName string, data map[string]interface{}) {
	if err := s.template.ExecuteTemplate(w, templateName, data); err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
}
