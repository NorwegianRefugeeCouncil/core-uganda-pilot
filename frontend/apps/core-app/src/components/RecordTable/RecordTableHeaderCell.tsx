import { HStack, IconButton, Pressable, Text } from 'native-base';
import { Icon } from 'core-design-system';
import React from 'react';
import { ColumnInstance, UseSortByColumnProps } from 'react-table';

import { RecordTableEntry } from './types';

type Props<T extends Record<string, any>> = {
  column: ColumnInstance & UseSortByColumnProps<T>;
};

export const RecordTableHeaderCell: React.FC<Props<RecordTableEntry>> = ({
  column,
}) => {
  const { onClick: handleOnClick } = column.getSortByToggleProps();
  const sortIcon = column.isSorted
    ? column.isSortedDesc
      ? 'more'
      : 'arrowUp'
    : 'arrowDown';

  return (
    <HStack
      width={column.width}
      p="2"
      borderBottomColor="neutral.300"
      borderBottomWidth="1"
      alignItems="center"
      justifyContent="space-between"
      bg="neutral.200"
    >
      {column.render(Text, {
        variant: 'body',
        fontWeight: '700',
        children: column.Header,
      })}
      {handleOnClick && (
        <IconButton
          onPress={(e) => handleOnClick(e)}
          colorScheme="secondary"
          variant="ghost"
          size="sm"
          icon={<Icon size={5} name={sortIcon} />}
        />
      )}
    </HStack>
  );
};
