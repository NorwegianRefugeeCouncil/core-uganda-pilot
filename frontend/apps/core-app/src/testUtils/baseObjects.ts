import { FormDefinition, FormType, Record } from 'core-api-client';

export const baseForm: FormDefinition = {
  id: 'form1',
  code: '',
  name: 'form 1',
  databaseId: 'databaseId1',
  folderId: 'folderId1',
  formType: FormType.DefaultFormType,
  fields: [],
};

export const baseTextField = {
  id: 'field1',
  name: 'field 1',
  description: 'description 1',
  code: '',
  required: false,
  key: false,
  fieldType: {
    text: {},
  },
};

export const baseMultilineTextField = {
  id: 'field2',
  name: 'field 2',
  description: 'description 2',
  code: '',
  required: true,
  key: true,
  fieldType: {
    multilineText: {},
  },
};

export const baseRecord1: Record = {
  id: 'record1',
  values: [
    {
      value: 'text',
      fieldId: 'field1',
    },
    {
      value: 'multi line text',
      fieldId: 'field2',
    },
  ],
  formId: 'form1',
  databaseId: 'databaseId1',
  ownerId: undefined,
};
