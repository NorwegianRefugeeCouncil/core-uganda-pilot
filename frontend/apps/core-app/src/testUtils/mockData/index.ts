export { makeRecord } from './Record';
export { makeForm } from './Form';
export { makeField } from './FormField';
