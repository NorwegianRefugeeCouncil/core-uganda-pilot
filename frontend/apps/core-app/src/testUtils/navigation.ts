export const mockNavigationProp = {
  dispatch: jest.fn(),
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getParent: jest.fn(),
  getState: jest.fn(),
  setParams: jest.fn(),
  setOptions: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
};
