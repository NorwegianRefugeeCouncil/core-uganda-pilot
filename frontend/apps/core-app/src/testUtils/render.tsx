import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import {
  render as r,
  RenderAPI,
  RenderOptions,
} from '@testing-library/react-native';
import { render as rWeb } from '@testing-library/react';
import { NativeBaseProvider } from 'native-base';
import { theme } from 'core-design-system';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

export const render = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
): RenderAPI =>
  r(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );

export const renderWeb = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
) =>
  rWeb(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );
