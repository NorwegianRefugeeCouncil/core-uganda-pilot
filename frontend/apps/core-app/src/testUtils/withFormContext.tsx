import { Button } from 'native-base';
import React from 'react';
import { FieldValues, FormProvider, useForm } from 'react-hook-form';

export const withFormContext =
  <T,>(
    Component: React.FC<T>,
    defaultValues: FieldValues = {},
    onSubmit: (data: any) => void = () => {},
  ) =>
  // eslint-disable-next-line react/display-name
  (props: T) => {
    const f = useForm({ defaultValues });

    const handleSubmit = f.handleSubmit((data: any) => {
      onSubmit(data);
    });

    return (
      <FormProvider {...f}>
        <Component {...props} />
        <Button testID="with-form-context-submit-button" onPress={handleSubmit}>
          Submit
        </Button>
      </FormProvider>
    );
  };
