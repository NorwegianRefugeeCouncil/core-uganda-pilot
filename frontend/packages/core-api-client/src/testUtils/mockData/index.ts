export * from './Form';
export * from './FormField';
export * from './Record';
