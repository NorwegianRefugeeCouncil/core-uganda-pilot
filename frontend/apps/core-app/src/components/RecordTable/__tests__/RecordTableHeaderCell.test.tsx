import React from 'react';

import { render } from '../../../testUtils/render';
import { RecordTableHeaderCell } from '../RecordTableHeaderCell';

it('should match the snapshot', () => {
  const { toJSON } = render(
    <RecordTableHeaderCell
      column={{
        getSortByToggleProps: () => ({ onClick: jest.fn() }),
        isSorted: true,
        isSortedDesc: false,
        render: jest.fn()
      }}
    />,
  );
  expect(toJSON()).toMatchSnapshot();
});
