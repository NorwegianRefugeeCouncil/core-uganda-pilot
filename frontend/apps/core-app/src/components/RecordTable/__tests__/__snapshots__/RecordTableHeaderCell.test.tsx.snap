// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#C7C7C7",
        "borderBottomColor": "#666666",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      dataSet={Object {}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onHoverIn={[Function]}
      onHoverOut={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        }
      }
    >
      <RNSVGSvgView
        accessibilityRole="image"
        align="xMidYMid"
        bbHeight={12}
        bbWidth={12}
        color={4280690214}
        dataSet={Object {}}
        fill="currentColor"
        focusable={false}
        meetOrSlice={0}
        minX={10}
        minY={10}
        stroke=""
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "color": "#262626",
              "height": 12,
              "width": 12,
            },
            Object {
              "flex": 0,
              "height": 12,
              "width": 12,
            },
          ]
        }
        tintColor={4280690214}
        vbHeight={20}
        vbWidth={20}
      >
        <RNSVGGroup
          fill={
            Array [
              2,
            ]
          }
          propList={
            Array [
              "fill",
              "stroke",
            ]
          }
          stroke={null}
        >
          <RNSVGGroup
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={
              Array [
                2,
              ]
            }
            strokeWidth="1.5"
          >
            <RNSVGPath
              d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
              fill={null}
              propList={
                Array [
                  "fill",
                  "strokeLinecap",
                ]
              }
              strokeLinecap={1}
            />
            <RNSVGPath
              d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
              fill={null}
              propList={
                Array [
                  "fill",
                  "strokeLinecap",
                ]
              }
              strokeLinecap={1}
            />
            <RNSVGCircle
              cx="12"
              cy="12"
              fill={null}
              propList={
                Array [
                  "fill",
                  "strokeMiterlimit",
                ]
              }
              r="11.25"
              strokeMiterlimit={10}
            />
          </RNSVGGroup>
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
